src/Symfony/Component/Mailer/Envelope.php

```diff
@@ -44,6 +44,9 @@ public static function create(RawMessage $message): self
 
     public function setSender(Address $sender): void
     {
+        if (preg_match('/.*[^\x00-\x7F]+.*@/', $sender->getAddress())) {
```

Comment:
maybe this?
`if (!preg_match('/^[^@\x80-\xFF]++@/', $sender->getAddress())) {`

Reply:
yes tested it a bit and seems to work nicely :+1: